container-disc/src/main/java/com/yahoo/container/jdisc/metric/GarbageCollectionMetrics.java

```diff
@@ -0,0 +1,93 @@
+package com.yahoo.container.jdisc.metric;
+
+import com.yahoo.jdisc.Metric;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.time.Clock;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Map;
+
+/**
+ * @author ollivir
+ */
+public class GarbageCollectionMetrics {
+    private static final String GC_COUNT = "jdisc.gc.count";
+    private static final String GC_TIME = "jdisc.gc.ms";
+    private static final String DIMENSION_KEY = "gcName";
+
+    public static final long REPORTING_INTERVAL = Duration.ofSeconds(62).toMillis();
+
+    static class GcStats {
+        private final long when;
+        private final long count;
+        private final long ms;
+
+        private GcStats(long when, long count, long ms) {
+            this.when = when;
+            this.count = count;
+            this.ms = ms;
```

Comment:
Is ms here a duration or an instant? I suggest to either make it a Duration or Instant, or simply add more context to the name.